test/functional/test_framework/script.py

```diff
@@ -689,6 +689,16 @@ def LegacySignatureHash(*args, **kwargs):
     else:
         return (hash256(msg), err)
 
+def sign_input_legacy(tx, input_index, input_scriptpubkey, privkey, sighash_type=SIGHASH_ALL):
```

Comment:
Is it worth checking things like the input index exists and sighash is only one byte?

Reply:
Seems like the input index is checked in `LegacySignatureMsg` (which is in turn called by `LegacySignatureHash`): https://github.com/bitcoin/bitcoin/blob/bc4f6b13feb29146b7e10e86f93dc7f6fb6937f2/test/functional/test_framework/script.py#L638-L639
For `sighash_type` it might be nice to add a type annotation, though I don't know if the byte range (0..255) can be specified exactly (currently that value is not set explicitly, as all our call-sites use the default SIGHASH_ALL). As for `tx.vin[input_index].scriptSig`: it's default-initialized with an empty byte-string b'' (see `CTxIn.__init__`), and if the user sets it to a `CScript` instance that's also okay, as this is derived from `bytes`, so I think we can safely assume that the type is okay.